flatpak/flatpak
# [1.12.x] Cherry-pick misc bug fixes from master

* tests: Allow FLATPAK_BINARY=flatpak for exports

    Author: @jtojnar, via #4496 

    (cherry picked from commit 96dbe28cfa96e80b23fa1d8072eb36edad41279c)

* tests: Generate Makefile-test-matrix.am.inc in $(srcdir)

    Author: me, via #4504

    (cherry picked from commit 647c51380c13b72adbd67e0aa83b3a2979f5c6a8)

* testlibrary: Don't use g_assert()

    Author: @mwleeds, via #4489  

    (cherry picked from commit 0258fc82bb8eac176d45781b54920f0a27613890)

* .gitignore: Update for recent changes

    Author: @mwleeds 
    
    (cherry picked from commit 2181f4f171abf824d8e3c754e9e7415813460856)

* tests: Add try-syscall helper

    Author: me, via #4505 

    (cherry picked from commit 4ce251882c488953ca6e3734f00c5dbe2e1e3e7a)

* tests: Add basic test coverage for our seccomp filters

    Author: me, via #4505 

    (cherry picked from commit 7c5aec474caef7aa004286cc9359611ad21d227b)

* flatpak-utils-http: Ensure to wake up the main context on error

    Author: @pwithnall, via #4518 

* run: Support PulseAudio socket path without unix: prefix

    Author: @wjt, via #4564 

    (cherry picked from commit 7534a970a5535d57d5fa07bd5ed9eb3647cbb7dc)

* run: Add link to PulseAudio server string documentation

    Author: @wjt, via #4564 

    (cherry picked from commit bcc114383b744e3dc32c0b859fc900f9793ce6c7)

* run: Document shortcomings of PulseAudio server string parsing

    Author: @wjt, via #4564 

    (cherry picked from commit eabbff6fefb879d6726a31afa8ae17ee144855a7)

* search: Don't strip .desktop suffix overzealously

    Author: @mwleeds, via #4536 

    (cherry picked from commit 62e09b406b8a7fb517c294d2c0b149a83f2cc64b)

* search: Use <bundle> ID to determine flatpak app ID

    Author: @mwleeds, via #4536 

    (cherry picked from commit 39de0ef280a98f67d639444cc6ea3bcfa61c0eec)

* app: Don't use carriage return for non-fancy output

    Author: @mwleeds, via #4532

    (cherry picked from commit 86d6918a1186467fb4e26408ec1bb947dcf7ba28)

* build-update-repo: Don't try to generate deltas of unknown refs

    Author: @mwleeds, via #4522

    (cherry picked from commit 17fbe516c4251c08a648f5e938af3af78fbd4ab4)

* tests: Use ${FLATPAK} not flatpak

    Author: @mwleeds, via #4580

    (cherry picked from commit d23793294d59125e645273d8b3518a9c19b597eb)

* Make test suite logs prettier

    Author: @mwleeds, via #4583

    (cherry picked from commit 388c23cfc51b6b9f03dcdf87b11c0078f11227ea)

* tests: Fix a comment

    Author: @mwleeds 
    
    (cherry picked from commit dfde010a49ed02c966ee9050e8ea4dc75033bdb6)

* app: Don't tab-complete on aliases

    Author: @mwleeds, via #4587 
    Fixes: #4036

    (cherry picked from commit 2c4c84ffee1a7b0c73f5af9dec02f63e6e1ef2ca)

* dir: Verify subsummary checksum from disk cache

    Author: @mwleeds, via #4568 
    Fixes: #4127

    (cherry picked from commit 6d74eec0a97171ef46fac7aa4017c235758a1f3d)

* Fix en_GB localization: Do not forget to pass the actual error message

    Author: @aleixpol, via #4606 

    (cherry picked from commit e28b1f31584a6fae71f51985187bc8f1755e9bd1)

* testlibrary: Tweak some helper functions

    Author: @mwleeds, via #4489 

    (cherry picked from commit 8f85f77ff9c1b1dea02ab96771a764ff86e7c024)

* testlibrary: Add missing cleanup

    Author: @mwleeds, via #4489 

    (cherry picked from commit f753dd214c9a44f462ca013e7357ead8f35538aa)

* testlibrary: Make remote existence assertions more friendly

    Author: @mwleeds, via #4489 

    (cherry picked from commit dd48e78652079556bebfe930caad2efa09656584)
